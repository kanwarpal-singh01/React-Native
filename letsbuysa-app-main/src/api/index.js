import {APIRequest} from './api-request';

export const API_LOGIN = 1;
export const API_SIGNUP = 2;
export const API_COUNTRY = 3;
export const API_VERIFY_CODE = 4;
export const API_FORGOT_PASSWORD = 5;
export const API_RESEND_OTP = 6;
export const API_GET_CATEGORIES = 7;
export const API_HOME_PAGE = 8;
export const API_VERIFY_MOBILE_CODE = 9;
export const API_LOGOUT = 10;
export const API_APP_CONFIG = 11;
export const API_CHANGE_NUMBER = 12;
export const API_GET_PRODUCTS = 13;
export const API_GET_PRODUCT_DETAIL = 14;
export const API_GET_WISHLIST = 15;
export const API_ADD_WISHLIST = 16;
export const API_VERIFY_FORGOT_CODE = 17;
export const API_CHANGE_FORGOT_PSWD = 18;
export const API_ADD_REVIEW = 19;
export const API_GET_ALL_REVIEWS = 20;
export const API_GET_ADDRESS = 21;
export const API_GET_COUNTRY = 22;
export const API_GET_STATES = 23;
export const API_DELETE_ADDRESS = 24;
export const API_GET_FILTERS = 25;
export const API_ADDRESS_UPDATE = 26;
export const API_ADD_ADDRESS = 27;
export const API_ADD_TO_CART = 28;
export const API_GET_CART = 29;
export const API_REMOVE_FROM_CART = 30;
export const API_MOVE_TO_WISHLIST = 31;
export const API_VERIFY_ADDRESS = 32;
export const API_VERIFY_ADDRESS_CODE = 33;
export const API_CHANGE_PSWD = 34;
export const API_UPDATE_PROFILE = 35;
export const API_SOCIAL_LOGIN = 36;
export const API_SOCIAL_REGISTER = 37;
export const API_CHECKOUT_CART = 38;
export const API_GET_ABOUT_INFO = 39;
export const API_APPLY_COUPON_CODE = 40;
export const API_APPLY_GIFT_CODE = 41;
export const API_GET_ORDERS = 42;
export const API_GET_ORDER_STATUS = 43;
export const API_GET_ORDER_DETAIL = 44;
export const API_ORDER_CANCEL = 45;
export const API_CHANGE_NOTIF_SETTINGS = 46;
export const API_GET_NOTIFICATIONS = 47;
export const API_GET_TRACKING_DETAILS = 48;
export const API_RETURN_ORDER = 49;
export const API_RETURN_ORDER_DETAIL = 50;
export const API_PAYMENT_STATUS = 51;
export const API_Wallet_STATUS = 52;
export const API_Wallet_Recharge_STATUS = 53;
export const API_Wallet_Recharge_Callback_STATUS = 54;
export const API_Voucher_STATUS = 55;
export const API_Brands_STATUS = 56;
export const API_Remove_checkout_STATUS = 57;
export const API_Move_checkout_STATUS = 58;
export const API_Remove_coupen_STATUS = 59;
export const API_Wallet_Apple_STATUS = 60;
export const API_CheckOut_Product_Check_STATUS = 61;
export const API_CheckOut_Apple_Pay_STATUS = 62;
export const API_CheckOutID_Apple_Pay = 63;
export const API_Wellet_CheckOutID_Apple_Pay = 64;






export const ApiURL = {
  signIn: '/login',  //
  signUp: '/register', //
  forgotPassword: '/forget', //
  verifyCode: '/otp_verify', //
  resendOTP: '/resend', //
  changeNumber: '/change_number', //
  verifyForgotCode: '/changepassword/verify', //
  changeForgotPassword: '/changepassword/update', //
  changePassword: '/changepassword', //
  logout: '/logout', //
  socialLogin: '/social/login', //
  socialRegister: '/social/register', //
  updateProfile: '/profile_update', //
  changeNotifSettings: '/profile_notification', //
  appConfig: '/appconfig', //
  getCategories: '/category', //
  homePage: '/home',
  getProducts: '/product', //
  getProductDetail: '/product/detail', //
  getWishlist: '/wishlist', //
  addToWishlist: '/wishlist/add', //
  addReview: '/product_review_add', //
  getAllReviews: '/product/review', 
  getAddress: '/address', //
  getCountries: '/country', //
  states: '/state', //
  getProductFilters: '/product/filter', 
  deleteAddressRequest: '/address/delete', //
  updateAddressRequest: '/address/update',
  addAddress: '/address/add', //
  addressVerify: '/address/send/code', //
  verifyAddressCode: '/address/verify/code', //
  addToCart: '/cart/add', //
  getCart: '/cart', //
  removeFromCart: '/cart/remove', //
  moveToWishlist: '/cart/movewishlist', //
  checkoutCart: '/checkout',
  checkoutNew: '/checkout/hyperpay',
  getAboutInfo: '/information',
  applyCoupon: '/coupon',
  applyGiftCode: '/voucher',
  getOrders: '/order', //
  getOrderStatus: '/order_status', //
  getOrderDetail: '/order/detail',
  orderCancel: '/ordercancel',
  getNotifications: '/notification', //
  getTrackingDetail: '/order/track',
  orderReturn: '/orderreturn/request',
  getReturnOrderDetail: 'orderreturn/detail',
  getPaymentStatus: 'ordercallback',
  wallet:'wallet',
  walletRecharg:'wallet/recharge',
  walletRechargeCallBack:'rechargecallback',
  voucher:'voucher',
  brands:'brands',
  removeProductCheckOut:'remove/product/checkout',
  moveProductCheckout:'movetowislit/product/checkout',
  removeCoupen:'remove_coupon',
  walletApple:'checkout/wallet/payment',
  productCheck:'check/product/availability',
  checkOutApplePay:'checkout/payment',
  checkOutIDApplePay:'checkout/apple-pay',

  checkWelletOutIDApplePay:'checkout/wallet/apple-pay',

  walletCheckOut:"get-wallet-url"



};

export {APIRequest};
