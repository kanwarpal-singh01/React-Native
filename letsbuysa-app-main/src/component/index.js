import CheckBox from 'src/component/ui/CheckBox'
import FloatingActionButton from 'src/component/ui/FloatingActionButton'
import FloatingInputText from 'src/component/ui/FloatingInputText'
import Label from 'src/component/ui/Label'
import PasswordInputText from 'src/component/ui/PasswordInputText'
import Ripple from 'src/component/ui/Ripple'
import RoundButton from 'src/component/ui/RoundButton'
import SquareButton from 'src/component/ui/SquareButton'
import CustomTabBar from 'src/component/ui/CustomTabBar';
import CodeInput from 'src/component/ui/ConfirmationCodeInput';
import Hr from 'src/component/ui/Hr';
import SocialRoundButton from 'src/component/ui/SocialRoundButton';
import Bounceable from 'src/component/ui/Bounceable';
import ModalDropdown from 'src/component/ui/ModalDropdown'
import InputDropdown from 'src/component/ui/InputDropDown';
import RadioComponent from 'src/component/ui/RadioComponent';

//views
import CustomNavigationHeader from 'src/component/view/CustomNavigationHeader/CustomNavigationHeader';
import SortPopUp from 'src/component/view/SortPopUp';
import ProductCard from 'src/component/view/ProductCard';
import LanguageChangePopUp from 'src/component/view/LanguageChangePopUp'
import FilterPopUp from 'src/component/view/FilterPopUp';
import ProductOptionPopUp from 'src/component/view/ProductOptionPopUp';
import CancelReasonPopUp from 'src/component/view/CancelReasonPopUp';

//Third party
import StepIndicator from 'src/component/ThirdParty/StepIndicator';

export {
    CheckBox,
    FloatingActionButton,
    FloatingInputText,
    Label,
    PasswordInputText,
    Ripple,
    RoundButton,
    SquareButton,
    CustomTabBar,
    CodeInput,
    Hr,
    SocialRoundButton,
    CustomNavigationHeader,
    SortPopUp,
    LanguageChangePopUp,
    ProductCard,
    Bounceable,
    FilterPopUp,
    ModalDropdown,
    InputDropdown,
    RadioComponent,
    ProductOptionPopUp,
    CancelReasonPopUp,
    StepIndicator
}