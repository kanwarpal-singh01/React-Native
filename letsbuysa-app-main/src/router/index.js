import React from 'react';
import {
    createStackNavigator,
    createAppContainer,
    NavigationActions,
} from 'react-navigation';

import Splash from 'src/screens/SplashScreen';
import VerifyCode from "src/screens/VerifyCode";
import ChangePhoneNumber from "src/screens/ChangePhoneNumber";
import ChangePassword from 'src/screens/ChangePassword';
import ProductDetail from "src/screens/ProductDetail";
import OrderDetail from "src/screens/OrderDetail";
import TrackOrder from 'src/screens/TrackOrder';
import ReturnOrder from 'src/screens/ReturnOrder';
import OrderTypeList from "src/screens/OrderTypeList";
import Cart from "src/screens/Cart";

import Routes from "./routes";
import MainRoute from "./MainRoute";

const navigator = createStackNavigator(
    {
        [Routes.Splash]: {
            screen: Splash
        },
        [Routes.MainRoute]: {
            screen: MainRoute
        },
        [Routes.VerifyCode]: {
            screen: VerifyCode,
        },
        [Routes.ChangePhoneNumber]: {
            screen: ChangePhoneNumber,
            navigationOptions: {
                header: null
            }
        },
        [Routes.ChangePswdDeepLink]: {
            screen: ChangePassword,
            navigationOptions: {
                header: null
            }
        },
        [Routes.ProductDetailDeeplink]: {
            screen: ProductDetail,
        },

        [Routes.ProductDetailNotif]: {
            screen: ProductDetail,
        },

        [Routes.OrderDetailNotif]: {
            screen: OrderDetail,
        },

        [Routes.TrackOrder]: {
            screen: TrackOrder,
        },

        [Routes.ReturnOrder]: {
            screen: ReturnOrder,
        },

        [Routes.ReturnOrderList]: {
            screen: OrderTypeList,
        },
        [Routes.CartNotif]: {
            screen: Cart,
        },
    },
    {
        initialRouteName: Routes.Splash,
        defaultNavigationOptions: {
            header: null
        }
    }
);

/*const DrawerNavigator = createDrawerNavigator({
    "Drawer1": {
        screen: Home
    },
    "Drawer2": {
        screen: Home
    },
    "Drawer3r": {
        screen: Home
    }
}, {
    drawerPosition: I18nManager.isRTL ? 'right' : 'left',
});*/


export const RootNavigator = createAppContainer(navigator);
