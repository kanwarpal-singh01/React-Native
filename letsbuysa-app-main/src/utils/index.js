import {Platform, Linking} from 'react-native';
import NetInfo from '@react-native-community/netinfo';
import {Color} from 'src/utils/Color';
import DateUtils from 'src/utils/DateUtils';
import Messages from 'src/utils/MessageUtils';
import Constants from 'src/utils/Constants';
import CommonStyle from 'src/utils/CommonStyles';
import ThemeUtils from 'src/utils/ThemeUtils';
import Strings from 'src/utils/strings';
import {validation, PasswordValidate} from 'src/utils/ValidationUtils';
import UtilityManager from 'src/utils/UtilityManager';
import SnackBarManager from 'src/utils/SnackBarManager';
import AdjustAnalyticsService from './AdjustAnalyticsService';

import fontelloConfig from '../../config.json';
import {createIconSetFromFontello} from 'react-native-vector-icons';
import {store} from 'src/redux/store';

const {
    Version,
    OS,
} = Platform;

export const IS_ANDROID = OS === 'android';
export const IS_IOS = OS === 'ios';
export const IS_LT_LOLLIPOP = Version < 21;


// use for check internet connection
/*export const isNetworkConnected = () => {
    if (Platform.OS === 'ios') {
        return new Promise(resolve => {
            const handleFirstConnectivityChangeIOS = isConnected => {
                NetInfo.isConnected.removeEventListener('change', handleFirstConnectivityChangeIOS);
                resolve(isConnected);
            };
            NetInfo.isConnected.addEventListener('change', handleFirstConnectivityChangeIOS);
        });
    }

    return NetInfo.isConnected.fetch();

};*/
const isNetworkConnected = async () => {
    let state = await NetInfo.fetch();
    /*console.log("Connection type", state.type);
    console.log("Is connected?", state.isConnected);*/
    return state.isConnected;
};
const Icon = createIconSetFromFontello(fontelloConfig);

const showErrorSnackBar = (message) => {
    let Snackbar = SnackBarManager.getSnackBar();
    let strings = store.getState().localeStrings ? store.getState().localeStrings : Strings;
    setTimeout(() => {
        Snackbar.alertWithType('error', strings.error, message, null, 800);
    }, 100);
};

const showSuccessSnackBar = (message) => {
    let Snackbar = SnackBarManager.getSnackBar();
    let strings = store.getState().localeStrings ? store.getState().localeStrings : Strings;
    setTimeout(() => {
        Snackbar.alertWithType('success', strings.success, message, null, 800);
    }, 100);
};

const showInfoSnackBar = (title, message, timeOut = 3000) => {
    let Snackbar = SnackBarManager.getSnackBar();
    setTimeout(() => {
        Snackbar.alertWithType('info', title, message, null, timeOut);
    }, 100);
};


const isCorrectType = (expected, actual) => {
    return Object.prototype.toString.call(actual).slice(8, -1) === expected;
};

const getValidArgumentsFromArray = function (array, type) {
    let validValues = [];
    array.forEach(function (value) {
        if (isCorrectType(type, value)) {
            validValues.push(value);
        }
    });

    return validValues;
};

const sendEmail = function (to, cc, bcc, subject, body, successCall, failCall) {
    let url = 'mailto:';
    let argLength = arguments.length;

    switch (argLength) {
        case 0:
            openLinkInBrowser(url);
            return;
        case 7:
            break;
        default:
            console.log('you must supply either 0 or 7 arguments. You supplied ' + argLength);
            return;
    }

    // we use this Boolean to keep track of when we add a new parameter to the querystring
    // it helps us know when we need to add & to separate parameters
    let valueAdded = false;

    if (isCorrectType('Array', arguments[0])) {
        let validAddresses = getValidArgumentsFromArray(arguments[0], 'String');

        if (validAddresses.length > 0) {
            url += encodeURIComponent(validAddresses.join(','));
        }
    }

    url += '?';

    if (isCorrectType('Array', arguments[1])) {
        let validAddresses = getValidArgumentsFromArray(arguments[1], 'String');

        if (validAddresses.length > 0) {
            valueAdded = true;
            url += 'cc=' + encodeURIComponent(validAddresses.join(','));
        }
    }

    if (isCorrectType('Array', arguments[2])) {
        if (valueAdded) {
            url += '&';
        }

        let validAddresses = getValidArgumentsFromArray(arguments[2], 'String');

        if (validAddresses.length > 0) {
            valueAdded = true;
            url += 'bcc=' + encodeURIComponent(validAddresses.join(','));
        }
    }

    if (isCorrectType('String', arguments[3])) {
        if (valueAdded) {
            url += '&';
        }

        valueAdded = true;
        url += 'subject=' + encodeURIComponent(arguments[3]);
    }

    if (isCorrectType('String', arguments[4])) {
        if (valueAdded) {
            url += '&';
        }

        url += 'body=' + encodeURIComponent(arguments[4]);
    }

    openLinkInBrowser(url);
};

const openLinkInBrowser = (link) => {
    let strings = store.getState().localeStrings ? store.getState().localeStrings : Strings;
    Linking.canOpenURL(link).then(supported => {
        if (supported) {
            Linking.openURL(link);
        } else {
            showInfoSnackBar(strings.information, strings.linkUnableToOpen);
        }
    }).catch(err => console.log('Error to open website', err));
};

const extractForgotCode = (link) => {
    link = link.replace(Constants.URL_REMOVE_REGEX, Constants.API_CONFIG.BASE_URL);
    let URLToDetect =
        Constants.API_CONFIG.BASE_URL +
        Constants.API_CONFIG.BASE_ROUTE.replace('restapi', '') +
        'account/reset&code';
    if (link.indexOf(URLToDetect) !== -1) {
        let i = link.indexOf('code=');
        let txt_length = 5;             //this is length of string "code="
        return link.substr(i + txt_length, link.length);
    }
    return null;
};

const extractProductSlug = (link) => {
    link = link.replace(Constants.URL_REMOVE_REGEX, Constants.API_CONFIG.BASE_URL);
    let URLToDetect = Constants.API_CONFIG.BASE_URL;
    if (link.indexOf(URLToDetect) !== -1) {
        let code = link.substr(URLToDetect.length, link.length);
        try {
            console.log('Slug Product code', code.split('?'), code);
            return code.split('?')[0];
        } catch (e) {
            return code;
        }
    }
    return null;
};

const decodeImageUrl = (val) => {
    let url = val ? val : Constants.THUMB_PLACEHOLDER;
    if (url && typeof url === 'string') {
        let thumb_url = url.replace(Constants.URL_REMOVE_REGEX, Constants.API_CONFIG.BASE_URL);
        return decodeURI(thumb_url);
    }
    return '';
};

const validateGeocodeAddress = (object) => {
    /*if ('address_line_1' in object && 'city' in object && 'country' in object) {
        return true
    }
    return false*/
    return true;
};

const handleReverseGeocodeAddress = (arrAddress) => {

    console.log('address will be formatted',arrAddress)
    for (let index in arrAddress) {

        let object = arrAddress[index];
        let showAddress = [];
        let formatedAdd = null
        let addressData = {};

        if(object.formattedAddress && !formatedAdd){
            formatedAdd = object.formattedAddress
        }

        if (object.country) {
            addressData['country'] = object.country;
        }

        if (object.locality) {
            addressData['city'] = object.locality;
        }

        if (object.adminArea && object.locality) {
            addressData['state'] = object.adminArea;
        } else if (object.adminArea) {
            addressData['city'] = object.adminArea;
        }

        if (object.streetName) {
            if (object.streetNumber) {
                addressData['address_line_2'] = object.streetNumber;
                showAddress.push(`${object.streetName},${object.streetNumber}`);
            } else {
                showAddress.push(object.streetName);
            }
            addressData['address_line_1'] = object.streetName;

            if (object.subLocality) {

                if (object.adminArea && object.adminArea !== object.subLocality) {
                    addressData['address_line_3'] = object.subLocality;
                }
            }
        } else if (object.subLocality) {
            addressData['address_line_1'] = object.subLocality;
            showAddress.push((object.subLocality));
        }

        if (object.subAdminArea) {
            if (!addressData['address_line_1']) {
                addressData['address_line_1'] = object.subAdminArea;
                showAddress.push(object.subAdminArea);
            } else if (!addressData['address_line_3']) {
                addressData['address_line_3'] = object.subAdminArea;
            }
        }

        if (object.postalCode) {
            addressData['pin_code'] = object.postalCode;
        }

        if (object.position) {
            addressData['latitude'] = object.position.lat;
            addressData['longitude'] = object.position.lng;
        }
        console.log(addressData);
        if (validateGeocodeAddress(addressData)) {
            console.log('Found Valide Address');
            return { fullAddress :formatedAdd, formatedAddress: showAddress.join(', '), searchAddressData: addressData};
        }
    }
    return undefined;
};

const isHexValid = (val) => {
    if (!val) {
        return false;
    }
    return /(^#[0-9A-F]{6}$)|(^#[0-9A-F]{3}$)/i.test(val);
};


const deepCompare = (x, y) => {
    let p, q, val = false;

    for (p in x) {
        if (x.hasOwnProperty(p) !== y.hasOwnProperty(p)) {
            return false;
        } else if (typeof x[p] !== typeof y[p]) {
            return false;
        }
    }

    for (p in y) {
        if (x.hasOwnProperty(p) !== y.hasOwnProperty(p)) {
            return false;
        } else if (typeof x[p] !== typeof y[p]) {
            return false;
        }
    }

    for (p in x) {
        for (q in y) {
            if (x[p] !== y[p]) {
                val = false;
                break;
            }
            val = true;
        }
        if (val === false) {
            return val;
        }
    }

    return val;
};

const capitalizeLetters = (string) => {
    let words = string.split(' '),
        newString = '';
    for (let i = 0; i < words.length; i++) {
        if (words[i] && words[i] !== ' ') {
            newString += words[i].charAt(0).toUpperCase() + words[i].substr(1).toLowerCase() + ' ';
        }
    }
    return newString;
};

const numberWithCommas = (x) => {
    return x?.toString().replace(/\B(?=(\d{3})+(?!\d))/g, ',');
};

const findLastIndex = (array, searchKey, searchValue) => {
    let index = array.slice().reverse().findIndex(x => x[searchKey] === searchValue);
    let count = array.length - 1;
    let finalIndex = index >= 0 ? count - index : index;
    console.log(finalIndex);
    return finalIndex;
};

export {
    Color,
    DateUtils,
    Messages,
    Constants,
    CommonStyle,
    ThemeUtils,
    Strings,
    UtilityManager,
    validation,
    PasswordValidate,
    isNetworkConnected,
    Icon,
    showErrorSnackBar,
    showSuccessSnackBar,
    showInfoSnackBar,
    SnackBarManager,
    openLinkInBrowser,
    extractForgotCode,
    decodeImageUrl,
    handleReverseGeocodeAddress,
    isHexValid,
    deepCompare,
    extractProductSlug,
    capitalizeLetters,
    numberWithCommas,
    findLastIndex,
    sendEmail,
    AdjustAnalyticsService,
};
