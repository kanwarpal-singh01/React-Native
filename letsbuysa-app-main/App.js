import React, {Component} from 'react';
import {StatusBar, View, Linking, NativeModules} from 'react-native';
import {Adjust, AdjustConfig} from 'react-native-adjust';
//Third party
import {Provider} from 'react-redux';
import {PersistGate} from 'redux-persist/integration/react';
import DropdownAlert from 'react-native-dropdownalert';
import {NavigationActions, StackActions} from 'react-navigation';
import firebase from 'react-native-firebase';
import type {
    Notification,
    NotificationOpen,
} from 'react-native-firebase/notifications';

//Utility
import {RootNavigator} from 'src/router';
import {store, persistor} from 'src/redux/store';
import {Constants, Color, SnackBarManager, IS_IOS, IS_ANDROID} from 'src/utils';
import CustomerSupportPopup from 'src/component/view/CustomerSupportPopUp';
import SearchPopUp from 'src/component/view/SearchPopUp';
import Routes from 'src/router/routes';
import NavigationRefManager from './src/utils/NavigationRefManager';
import Actions from './src/redux/action';



export default class App extends Component {
    //Utility
    _handleOpenURL = event => {
        let link = event.url,
            action = null;
        console.log('Link from normal deeplink==>', link);
        if (link.indexOf('https') === -1) {
            link = link.replace('http', 'https');
        }
        let URLToDetect =
            Constants.API_CONFIG.BASE_URL +
            Constants.API_CONFIG.BASE_ROUTE.replace('restapi', '') +
            'account/reset&code',
            URLToDismiss =
                Constants.API_CONFIG.BASE_URL +
                Constants.API_CONFIG.BASE_ROUTE.replace('restapi', '');

        if (link.indexOf(URLToDetect) !== -1) {
            action = NavigationActions.navigate({
                routeName: Routes.ChangePswdDeepLink,
                params: {
                    forgotLink: link,
                    appState: 'background',
                },
            });
        } else if (link.indexOf(URLToDismiss) !== -1) {
            action = null;
        } else if (
            link.indexOf(Constants.API_CONFIG.BASE_URL) !== -1 &&
            link !== Constants.API_CONFIG.BASE_URL
        ) {
            action = NavigationActions.navigate({
                routeName: Routes.ProductDetail,
                params: {
                    productSlugLink: link,
                    appState: 'background',
                },
            });
        }

        if (action) {
            this.navigationRef && this.navigationRef.dispatch(action);
        }
    };

    handleOpenNotification = notificationData => {
        firebase.notifications().removeAllDeliveredNotifications();
        switch (notificationData.type) {
            case 'order':
                if (
                    notificationData.order_id &&
                    this.navigationRef &&
                    store.getState().user
                ) {
                    this.navigationRef.dispatch(
                        StackActions.push({
                            routeName: Routes.OrderDetailNotif,
                            params: {
                                orderData: {order_id: notificationData.order_id},
                                fromSplash: false,
                            },
                        }),
                    );
                }
                break;
            case 'return_order':
                if (
                    notificationData.order_id &&
                    this.navigationRef &&
                    store.getState().user
                ) {
                    this.navigationRef.dispatch(
                        StackActions.push({
                            routeName: Routes.ReturnOrderList,
                            params: {
                                currentOrderType: {
                                    key: '7',
                                    type: Constants.OrderTypeReturn,
                                },
                                fromRoute: Routes.MainRoute,
                            },
                        }),
                    );
                }
                break;
            case 'product':
                if (
                    notificationData.product_id &&
                    this.navigationRef &&
                    store.getState().user
                ) {
                    this.navigationRef.dispatch(
                        StackActions.push({
                            routeName: Routes.ProductDetailNotif,
                            params: {
                                productData: {product_id: notificationData.product_id},
                                fromSplash: false,
                            },
                        }),
                    );
                }
                break;
            case 'cart':
                if (this.navigationRef && store.getState().user) {
                    this.navigationRef.dispatch(
                        StackActions.push({
                            routeName: Routes.CartNotif,
                            params: {
                                fromSplash: false,
                            },
                        }),
                    );
                }
                break;
        }
    };

    setRefs = navigatorRef => {
        this.navigationRef = navigatorRef;
        NavigationRefManager.setTopLevelNavigator(navigatorRef);
    };

    getToken = async () => {
        let fcmToken = store.getState().fcmToken;
        console.log(fcmToken);
        if (!fcmToken) {
            fcmToken = await firebase.messaging().getToken();
            if (fcmToken) {
                store.dispatch(Actions.setFCMToken(fcmToken));
            }
        }
    };

    checkPermission = async () => {
        const enabled = await firebase.messaging().hasPermission();
        if (enabled) {
            this.getToken();
        } else {
            this.requestPermission();
        }
    };

    requestPermission = async () => {
        try {
            await firebase.messaging().requestPermission();
            this.getToken();
        } catch (error) {
            console.log('permission rejected');
            this.getToken();
        }
    };

    createNotificationListeners = async () => {
        firebase.notifications().onNotification(notification => {
            notification.android
                .setChannelId('default_letsbuy_notification')
                .setSound('default');
            notification.android.setSmallIcon('ic_stat_ic_notification');
            notification.android.setAutoCancel(true);
            notification.android.setColor(Color.PRIMARY);
            firebase
                .notifications()
                .displayNotification(notification)
                .catch(err => {
                    console.log(err);
                });
        });

        this.removeNotificationOpenedListener = firebase
            .notifications()
            .onNotificationOpened((notificationOpen: NotificationOpen) => {
                // Get the action triggered by the notification being opened
                const action = notificationOpen.action;
                // Get information about the notification that was opened
                const notification: Notification = notificationOpen.notification;
                if (notification && notification.data) {
                    console.log(notification.data);
                    this.handleOpenNotification(notification.data);
                }
            });
    };

    constructor(props) {
        super(props);
        this.navigationRef = null;
        const adjustConfig = new AdjustConfig(
            Constants.ADJUST_TRACKING.APP_TOKEN,
            AdjustConfig.EnvironmentProduction,
        );
        // adjustConfig.setShouldLaunchDeeplink(true)
        adjustConfig.setDeferredDeeplinkCallbackListener(deepLink => {
            console.log('Deferred deep link URL content: ' + deepLink);
            this._handleOpenURL({url:deepLink})
        });
        adjustConfig.setLogLevel(AdjustConfig.LogLevelVerbose);
        Adjust.create(adjustConfig);
    }

    componentDidMount() {
        setTimeout(
            () => {
                Linking.addEventListener('url', this._handleOpenURL);
            },
            IS_IOS ? 1000 : 0,
        );

        const channel = new firebase.notifications.Android.Channel(
            'default_letsbuy_notification',
            'App notifications',
            firebase.notifications.Android.Importance.Max,
        );

        firebase.notifications().android.createChannel(channel);
        this.checkPermission();
        this.createNotificationListeners();
    }

    componentWillUnmount() {
        Linking.removeEventListener('url', this._handleOpenURL);
        this.removeNotificationOpenedListener();
        IS_ANDROID && NativeModules.GetDeeplink.setOpenedUrl('');
        Adjust.componentWillUnmount();
    }

    render() {
        return (
            <View style={{flex: 1}}>
                <Provider store={store}>
                    <PersistGate loading={null} persistor={persistor}>
                        <StatusBar
                            barStyle={'light-content'}
                            backgroundColor={Color.BG_COLOR_DARK}
                        />
                        <RootNavigator ref={this.setRefs}/>
                        <CustomerSupportPopup/>
                        <SearchPopUp/>
                    </PersistGate>
                </Provider>
                <DropdownAlert
                    ref={ref => SnackBarManager.setSnackBar(ref)}
                    useNativeDriver={true}
                    titleStyle={{
                        fontFamily: Constants.FontStyle.bold,
                        color: Color.WHITE,
                    }}
                    messageStyle={{
                        fontFamily: Constants.FontStyle.regular,
                        color: Color.WHITE,
                    }}
                    inactiveStatusBarStyle={'light-content'}
                    inactiveStatusBarBackgroundColor={Color.TEXT_DARK}
                />
            </View>
        );
    }
}
