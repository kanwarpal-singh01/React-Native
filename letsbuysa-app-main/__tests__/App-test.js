/**
 * @format
 */

import 'react-native';
import React from 'react';
import App from '../App';

import {Label} from 'src/component';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';


jest.mock('react-navigation', () => {
    return {
        createAppContainer: jest.fn().mockReturnValue(function NavigationContainer(props) {
            return null;
        }),
        createDrawerNavigator: jest.fn(),
        createBottomTabNavigator: jest.fn(),
        createStackNavigator: jest.fn(),
        StackActions: {
            push: jest.fn().mockImplementation(x => ({...x, "type": "Navigation/PUSH"})),
            replace: jest.fn().mockImplementation(x => ({...x, "type": "Navigation/REPLACE"})),
        },
        NavigationActions: {
            navigate: jest.fn().mockImplementation(x => x),
        }
    }
});

jest.mock('redux-persist/integration/react', () => ({
    PersistGate: props => props.children,
}));

it('label renders correctly', () => {
    const tree = renderer.create(<Label/>).toJSON();
    expect(tree).toMatchSnapshot();
});
