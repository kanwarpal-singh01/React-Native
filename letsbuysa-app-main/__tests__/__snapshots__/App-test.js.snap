// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`label renders correctly 1`] = `
<Text
  numberOfLines={null}
  onPress={null}
  style={
    Array [
      Object {
        "fontSize": 21,
      },
      Object {
        "fontWeight": "normal",
      },
      Object {
        "fontFamily": "Roboto-Regular",
      },
      Object {
        "color": "#445c6d",
        "marginBottom": 0,
        "marginLeft": 0,
        "marginRight": 0,
        "marginTop": 0,
        "textAlign": "left",
      },
      undefined,
    ]
  }
/>
`;
